pkg/host/store.go

```diff
@@ -211,6 +211,9 @@ func (s *Store[E]) Watch(ctx context.Context) (store.Watch[E], error) {
 }
 
 func (s *Store[E]) get(id string) (E, error) {
+	s.idMu.RLock(id)
```

Comment:
it cannot work. You will create deadlock in update function. In update function you locked mutex for writting so RLock will wait here for unlock of write operation which cannot be done because it will be block RLock function.

Reply:
No, it won't lead to deadlock. as get will increase the redercount and write will wait for it to complete and then write. I validated it. It's working fine.

It would have led to deadlock in current implementations due to the reason you mentioned.

Reply:
but you can create starvation problem in this case, i think. if you will get very quickly new reading operation, write can be postponed forever.

Reply:
In theory, it's possible. However, I haven't encountered such an issue even after running multiple instances of consecutive integration tests, especially considering the already super-fast read operations. Therefore, I believe a client sending GET requests faster than the this might not be a practical scenario.

However, we can always refactor if we see any such situation in real-time in future.

Reply:
As discussed offline, there's no need to over-engineer in this case. Therefore, proceeding with a simple solution in the latest commit.